docs/connectors/sinks/custom-sinks.md

```diff
@@ -0,0 +1,123 @@
+# Creating a Custom Sink
+
+Quix Streams provides basic facilities to implement custom sinks for external destinations (currently in beta).
+
+To create a new sink, extend and implement the following Python base classes:
+- `quixstreams.sinks.base.sink.BaseSink` - the parent interface for all sinks.  
+The `StreamingDataFrame.sink()` accepts implementations of this class.
+
+- `quixstreams.sinks.base.sink.BatchingSink` - a base class for batching sinks, that need to batch data first before writing it to the external destination.  
+Check out [InfluxDBV3Sink](influxdb3-sink.md) and [CSVSink](csv-sink.md) for example implementations of the batching sinks.
+
+
+Here is the code for `BaseSink` class for the reference:
+
+```python
+import abc
+
+class BaseSink(abc.ABC):
+    """
+    This is a base class for all sinks.
+
+    Subclass and implement its methods to create your own sink.
+
+    Note that Sinks are currently in beta, and their design may change over time.
+    """
+
+    @abc.abstractmethod
+    def flush(self, topic: str, partition: int):
+        """
+        This method is triggered by the Checkpoint class when it commits.
+
+        You can use `flush()` to write the batched data to the destination (in case of
+        a batching sink), or confirm the delivery of the previously sent messages
+        (in case of a streaming sink).
+
+        If flush() fails, the checkpoint will be aborted.
+        """
+
+    @abc.abstractmethod
+    def add(
+        self,
+        value: Any,
+        key: Any,
+        timestamp: int,
+        headers: List[Tuple[str, HeaderValue]],
+        topic: str,
+        partition: int,
+        offset: int,
+    ):
+        """
+        This method is triggered on every new record sent to this sink.
+
+        You can use it to accumulate batches of data before sending them outside, or
+        to send results right away in a streaming manner and confirm a delivery later
+        on flush().
+        """
+
+    def on_paused(self, topic: str, partition: int):
+        """
+        This method is triggered when the sink is paused due to backpressure, when
+        the `SinkBackpressureError` is raised.
+
+        Here you can react to backpressure events.
+        """
+```
+
+
+## Sinks workflow
```

Comment:
It makes sense why you put this here, but I wonder if this and (maybe) the performance section maybe belongs in the main sinks README?

Reply:
Agree, makes sense